addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/application_error_scanner_regex.js

```diff
@@ -0,0 +1,73 @@
+            "&lt;font style=\"COLOR: black; FONT: 8pt/11pt verdana\"&gt;\\s+(\\[Macromedia\\]\\[SQLServer\\sJDBC\\sDriver\\]\\[SQLServer\\]|Syntax\\serror\\sin\\sstring\\sin\\squery\\sexpression\\s)",
+            "(?i)The Error Occurred in &lt;b&gt;(.*): line.*&lt;\/b&gt;&lt;br&gt;",
+            "(?i)The error occurred while processing.*Template: (.*) &lt;br&gt;.",
+            "(?i)The error occurred while processing.*in the template file (.*)\\.&lt;\/p&gt;&lt;br&gt;",
+            "(?i)&lt;span&gt;&lt;H1&gt;Server\\sError\\sin\\s'[^']*'\\sApplication\\.&lt;hr\\swidth=100%\\ssize=1\\scolor=silver&gt;&lt;/H1&gt;",
+            "(?i)&lt;title&gt;Invalid\\sfile\\sname\\sfor\\smonitoring:\\s'([^']*)'\\.\\sFile\\snames\\sfor\\smonitoring\\smust\\shave\\sabsolute\\spaths\\,\\sand\\sno\\swildcards\\.&lt;/title&gt;",
+            "(?i)&lt;b&gt;(Warning|Fatal\\serror|Parse\\serror)&lt;/b&gt;:\\s+.*?\\sin\\s&lt;b&gt;.*?&lt;/b&gt;\\son\\sline\\s&lt;b&gt;\\d*?&lt;/b&gt;&lt;br\\s/&gt;",
+            "(?:Unknown database '.*?')|(?:No database selected)|(?:Table '.*?' doesn't exist)|(?:You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '.*?' at line .*?)",
+            "Exception report.*message.*description.*exception.*note.*",
+            "(?i)&lt;head&gt;&lt;title&gt;JRun Servlet Error&lt;/title&gt;&lt;/head&gt;",
+            "(?i)&lt;h1&gt;Servlet\\sError:\\s\\w+?&lt;/h1&gt;",
+            "(?i)Servlet\\sError&lt;/title&gt;"];
+
+var errorPatterns = [];
+patterns.forEach(function(pattern){
+    errorPatterns.push(java.util.regex.Pattern.compile(pattern));
+});
+
+function scan(helper,msg) {
+
+    if(msg.opcode != OPCODE_TEXT || msg.isOutgoing){
+        return;
+    }
+
+    var counter = 0;
+    var message = String(msg.getReadablePayload()).valueOf();
+
+    errorPatterns.forEach(function(pattern){
+        var matcher = pattern.matcher(message);
+        while(matcher.find()){
+            helper.newAlert()
+                .setRiskConfidence(RISK_MEDIUM, CONFIDENCE_MEDIUM)
+                .setName("Application Error Disclosure via WebSockets (regex-script)")
+                .setDescription("This page contains an error/warning message that \
+ may disclose sensitive information like the location of the file \
+ that produced the unhandled exception. This information can be used\
+ to launch further attacks against the web application. The alert\
+ could be a false positive if the error message is found inside a\
+ documentation page.")
+                .setSolution("Review the source code of this page. Implement custom \
+ error pages. Consider implementing a mechanism to provide a unique\
+ error reference/identifier to the client (browser) while logging the\
+ details on the server side and not exposing them to the user.")
+                .setParam(counter++)
```

Comment:
I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

Reply:
Fixed in 319f60c